app/models/Backend.scala

```diff
@@ -783,6 +808,29 @@ class Backend @Inject() (implicit
     }
   }
 
+  def filterLiteratureByDate(
```

Comment:
I think this signature can be shorter. `date` and `dateComparator` are both `Option`, but it only makes sense to have one if both fields are of type `Some`. We can wrap an tuple in an `Option` so that we can do a single match on whether we're filtering by date or not:
```
  def filterLiteratureByDate(pub: Publication,
                             filter: Option[(Long, ComparatorEnum.Comparator)]): Boolean = {
    filter match {
      case Some(dataAndComparator) => ???
      case None        => true
    }
```